server/v2/commands.go

```diff
@@ -65,8 +82,8 @@ func Commands(logger log.Logger, homePath string, modules ...ServerModule) (CLIC
 	return cmds, nil
 }
 
-func AddCommands(rootCmd *cobra.Command, logger log.Logger, homePath string, modules ...ServerModule) error {
-	cmds, err := Commands(logger, homePath, modules...)
+func AddCommands(rootCmd *cobra.Command, appCreator NewServerModulesFunc, codec transaction.Codec[transaction.Tx], logger log.Logger, homePath string, modules ...ServerModule) error {
```

Comment:
why does it need to take a codec?

Reply:
for cometbft initialization I think

Reply:
but that is done within the cometbft server instantiation, which is a servermodule, so it just gets passed initiated in the variadic.

Reply:
should we init server modules in `newRootCmd` or in server start cmd. 
Im following 2nd way, pass empty modules to `AddCommands` then get their default config to write. The actual modules initialization process when the user starts the server then set to server.modules

wdut @julienrbrt

Reply:
You decide what server modules you are going to use in your app.

We manage the config writing, starting and stopping of the servers.

So the server part should be as generic as possible

Reply:
For the configs we need to read it in start, but the init shouldn't still be done before. Which means we need to change a bit the current constructors of the servers

Reply:
How about making server.New() as an interface that take viper, logger, app as input. So user just define empty modules that they want to use (write default config in this step). Then in start, modules is initialized with right config
temp approach https://github.com/cosmos/cosmos-sdk/commit/6eda0c7a85327c398ee774a98d275ee627214578